### PR TITLE
[Community] add fury release documentation

### PR DESCRIPTION
**Pull request by chaokunyang**
# What this PR do?

This release doc is referenced from opendal and streampark, thanks for their great work.

The website screencut: 

image
image

# Related issue
https://github.com/apache/incubator-fury/issues/1421
https://github.com/apache/incubator-fury/issues/1389

### CODE REVIEW

**Comment by chaokunyang** [comment]
Hi @tisonkun @pjfanning  @vesense @PragmaTwice @theweipeng @eolivelli  @carp84 I'm not sure whether all the process are correct, could you help review this release process.

File: docs/community/how_to_release.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+title: How to release
+sidebar_position: 0
+id: how_to_release
+---
+
+This document mainly introduces how the release manager releases a new version of Apache Fury.
+
+## Introduction
+
+Source Release is the msot important part which Apache values.
```
Comment (pjfanning) [inline docs/community/how_to_release.md:11]: Typo in msot

Comment (chaokunyang) [reply docs/community/how_to_release.md]: thanks, fixed

File: docs/community/how_to_release.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+svn add .   # It is not needed if the KEYS document exists before.
+svn ci -m "add gpg key for YOUR_NAME" # Later on, if you are asked to enter a username and password, just use your apache username and password.
+```
+
+#### Upload the GPG public key to your GitHub account
+- Enter https://github.com/settings/keys to add your GPG key.
+- Please remember to bind the email address used in the GPG key to your GitHub account (https://github.com/settings/emails) if you find "unverified" after adding it.
+
+### Further reading
+It's recommanded but not mandatory to read following documents before making a release to know more details about apache release:
```
Comment (pjfanning) [inline docs/community/how_to_release.md]: Typo in recommanded

File: docs/community/how_to_release.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+
+Thanks for reviewing and voting for our release candidate.
+
+We will proceed with publishing the approved artifacts and sending out the announcement soon.
+```
+
+### What if vote fail
+
+If the vote failed, click "Drop" to drop the staging Maven artifacts.
+
```
Comment (chaokunyang) [inline docs/community/how_to_release.md:446]: If there are some concerns raised from the vote thread, could we update current build artifacts  without filing a new vote thread if we don't update the code. 

For example, one may forget add signature by accident. Could we continute current vote if the release manager uploaded the signature file and reply in the vote thread

File: docs/community/how_to_release.md (hunk 2 above)
Comment (chaokunyang) [reply docs/community/how_to_release.md]: fixed

File: docs/community/how_to_release.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+id: how_to_release
+---
+
+This document mainly introduces how the release manager releases a new version of Apache Fury.
+
+## Introduction
+
+Source Release is the msot important part which Apache values.
+
+Plese pay more attention to license and signing issues.
```
[the remark this answers is not in the document]
Comment (chaokunyang) [reply docs/community/how_to_release.md]: fixed

File: docs/community/how_to_release.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+#### Check whether the key is created successfully
+Uploading takes about one minute; after that, you can check by your email at the corresponding keyserver.
+
+Uploading keys to the keyserver is mainly for joining a [Web of Trust](https://infra.apache.org/release-signing.html#web-of-trust).
+
+#### Add your GPG public key to the project KEYS file
+
+The svn repository of the release branch is: https://dist.apache.org/repos/dist/release/incubator/fury
+
+Please add the public key to KEYS in the release branch:
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply docs/community/how_to_release.md:135]: Apologies - should be dev not staging

[the remark this answers is not in the document]
Comment (chaokunyang) [reply docs/community/how_to_release.md:135]: I see, thanks. This will be much better

[the remark this answers is not in the document]
Comment (chaokunyang) [reply docs/community/how_to_release.md:135]: You mean something like this:
```bash
svn checkout --depth empty https://dist.apache.org/repos/dist apache-dist
cd apache-dist
svn update --set-depth infinity dev/incubator/fury release/incubator/fury
```

[the remark this answers is not in the document]
Comment (chaokunyang) [reply docs/community/how_to_release.md:135]: Will `svn mv https://dist.apache.org/repos/dist/dev/incubator/fury/${release_version}-${rc_version} https://dist.apache.org/repos/dist/release/incubator/fury/${release_version} -m "Release fury ${release_version}"` work directly without pull it to local?

**Comment by chaokunyang** [comment]
Hi, with two versions released, I think this PR is ready for review

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**